run_flax_speech_recognition_seq2seq_streaming_review.py

```diff
@@ -0,0 +1,1391 @@
+        eval_step, label_smoothing_factor=training_args.label_smoothing_factor), "batch")
+    p_generate_step = jax.pmap(generate_step, "batch")
+
+    # Replicate the train state on each device
+    state = state.replicate()
+    
+    # Logging
+    logger.info("***** Running training *****")
+    logger.info(
+        f"  Original model = {model_args.model_name_or_path}")
+    if training_args.push_to_hub:
+        logger.info(
+        f"  Hub model id = {training_args.hub_model_id}")
+    logger.info(
+        f"  Dataset name = {data_args.dataset_name}")
+    logger.info(
+        f"  Dataset config name = {data_args.dataset_config_name}")
+    logger.info(
+        f"  Learning rate = {training_args.learning_rate}")
+    logger.info(
+        f"  Scheduler = {training_args.lr_scheduler_type}")
+    logger.info(
+        f"  Num examples = {data_args.num_train_steps * train_batch_size:,}")
+    if model_args.num_beams:
+        logger.info(
+        f"  Num beams evaluation = {model_args.num_beams}")
+    logger.info(
+        f"  Number of hosts = {num_of_hosts}")
+    if num_of_hosts > 1:
+        logger.info(
+            f"  Current host idx = {current_host_idx}")
+    logger.info(
+        f"  Instantaneous batch size per device = {training_args.per_device_train_batch_size}")
+    logger.info(
+        f"  Total train batch size per node (w. parallel & distributed) = {train_batch_size // num_of_hosts:,}")
+    logger.info(
+        f"  Total train batch size (w. parallel & distributed) = {train_batch_size}")
+    if training_args.gradient_accumulation_steps > 1:
+        logger.info(
+            f"  Gradient accumulation steps = {training_args.gradient_accumulation_steps}")
+        logger.info(f"  ↪ Effective total batch size = {train_batch_size * training_args.gradient_accumulation_steps:,}")
+    logger.info(f"  Total optimization steps = {data_args.num_train_steps - training_state['step']:,}")
+    if training_state['step'] > 0:
+        logger.info(f"  ↪ Starting at {training_state['step']:,} and finishing at {data_args.num_train_steps:,}")
+
+    if model_args.dropout or model_args.attention_dropout or model_args.activation_dropout or model_args.encoder_dropout or model_args.decoder_dropout:
+        logger.info("  Dropout = True")
+        if model_args.dropout:
+            logger.info(f"  ↪ Dropout probability = {model_args.dropout}")
+        if model_args.attention_dropout:
+            logger.info(f"  ↪ Attention dropout probability = {model_args.attention_dropout}")
+        if model_args.activation_dropout:
+            logger.info(f"  ↪ Activation dropout probability = {model_args.activation_dropout}")
+        if model_args.encoder_dropout:
+            logger.info(f"  ↪ Encoder dropout probability = {model_args.encoder_dropout}")
+        if model_args.decoder_dropout:
+            logger.info(f"  ↪ Decoder dropout probability = {model_args.decoder_dropout}")
+
+    train_time = 0
+
+    # Training summary
```

Comment:
Super nice way of holding your training attributes - but unfortunately probably too much for a MWE in transformers, sorry!